design-documents/testing/functional/test-package.md

```diff
@@ -0,0 +1,96 @@
+# Magento Functional Test packaging
+The purpose of this document is to define a strategy of functional test packaging.
+
+## Terminology
+`MFTF` - Magento Functional Testing Framework
+
+`MFTF tests` - means tests written to cover Magento functionality using MFTF
+
+## Overview
+### Current situation
+In git repo all MFTF tests currently located under `app/code/Magento/*/Test/Mftf` directory.
+When Magento releasing new version we package all modules to own packages and distribute them via repo.magento.com.
+It means that all MFTF tests currently also distributed as a part of `magento2-module`.
+When SIs or Extension developers installs Magento via Composer they are able to run and use MFTF tests for their own needs.
+
+### Problem
+This approach leads to situation where production environment always contains code (in our case xml configurations) which will never be used.
+
+Also here is reasons described by Piotr Kaminski why it is not good to have tests on prod env.
+>There are couple reasons to separate tests from core distribution:
+>
+>Security:
+> - reduce possible attack surface (less code)
+> - the test code is not written with security in mind
+> - even if we protect the access via .htaccess, some sites run Nginx or skip .htaccess making this code accessible
+> - we had security issues in the past that resulted in remote code exec using code related to tests
+>
+>Future-proof:
+> - ability to release separately, on separate cadence
+> - matches the general plan for all other non-critical modules - we want to modularize and release separately things like payment extensions, shipping and more.
+
+
+### Proposed solution
+During Magento package new releases automated tool will exclude tests from main `magento2-module` package and create other package with type `magento2-test-module`.
+
+## Packaging flows
+### General Flow
+1. Exclude `Test` folder from module and keep it for further packaging flow as it shown on image below.
+
+![Packaging Process](img/packaging.png)
```

Comment:
This is different from the plan shown here where the Test/MFTF has only composer.json: https://devdocs.magento.com/guides/v2.3/extension-dev-guide/build/module-file-structure.html#module-file-structure

The original plan assumed that MFTF can run independently outside of an M2 instance, hence only this needs to be packaged separately.

Reply:
Hi @ravmenon 
Yes, this picture is a bit outdated. I'll update it to show that MFTF directory is going to be packaged only.